src/Snackbar/SnackbarContainer.tsx

```diff
@@ -0,0 +1,74 @@
+import React, {
+  createContext,
+  FC,
+  useState,
+  useCallback,
+  ReactNode,
+  useContext,
+} from 'react'
+import styled from 'styled-components'
+import { SnackbarItem } from './SnackbarItem'
+import { CreateSnack, Snackbar, SnackbarContextType } from './types'
+
+export const SnackbarContext = createContext<SnackbarContextType>({
+  addSnackbar: () => {
+    throw new Error('Please add the SnackbarContainer to your application')
+  },
+})
+
+export const useSnackbarContext = () => useContext(SnackbarContext)
+
+export const SnackbarContainer: FC<{ children?: ReactNode }> = ({
+  children,
+}) => {
+  let snackbarId = 0
+  const [snackbars, setSnackbars] = useState<Snackbar[]>([])
+
+  // create snackbar unique ID
+  const newSnackId = useCallback(() => {
+    const newId = `MM_SNACKBAR_${snackbarId}`
```

Comment:
IIRC we have a unique ids generator in utils

Reply:
@fleonard I did try and re-use it but due to it being a custom hook, you can't use it within the `addSnackbar` function 😢

Reply:
oh damn - fair enough!